dev/io.openliberty.checkpoint_fat_wsoc/fat/src/io/openliberty/checkpoint/fat/BasicTest.java

```diff
@@ -0,0 +1,222 @@
+
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.StringBuilder;
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import org.asynchttpclient.Dsl;
+import org.asynchttpclient.ws.WebSocket;
+import org.asynchttpclient.ws.WebSocketListener;
+import org.asynchttpclient.ws.WebSocketUpgradeHandler;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.CheckpointTest;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+/**
+ * Basic WebSockets tests to ensure the behavior works as expected after a checkpoint restore. 
+ */
+@RunWith(FATRunner.class)
+@CheckpointTest
+public class BasicTest {
+
+    @Server("basicWsocServer")
+    public static LibertyServer server;
+
+    private static final Logger LOG = Logger.getLogger(BasicTest.class.getName());
+
+    private static final String APP_NAME = "basic";
+
+    private static Boolean IS_EXPECTED_RESULT = false;
+
+    private static CountDownLatch latch; // Used to address timing issues between when the message is recived and when the assert is checked
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        // Build the war app and add the dependencies
+        ShrinkHelper.defaultDropinApp(server, APP_NAME + ".war", "io.openliberty.wsoc.basic");
+
+        server.setCheckpoint(CheckpointPhase.AFTER_APP_START, true, null);
```

Comment:
Calling `setCheckpoint` twice was probably not intended.  The second call below will override what this call did.

Reply:
Fixed! Thanks!